Add a docs link to the CockroachDB dialect

rafiss opened the pull request:
<!-- Provide a general summary of your proposed changes in the Title field above -->

### Description
<!-- Describe your changes in detail -->

This is a docs only change to point to the sqlalchemy-cockroachdb dialect.

### Checklist
<!-- go over following points. check them with an `x` if they do apply, (they turn into clickable checkboxes once the PR is submitted, so no need to do everything at once)

-->

This pull request is:

- [x] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [ ] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

rafiss committed:
Add a docs link to the CockroachDB dialect

CaselIT commented on doc/build/dialects/index.rst:
```diff
@@ -43,47 +43,49 @@ External Dialects
++---------------------------------------+-------------------------+
+| Apache Solr                           | sqlalchemy-solr_        |
++---------------------------------------+-------------------------+
+| CockroachDB                           | sqlalchemy-cockroachdb_ |
```
I just noticed that the whole table needed to be reformatted.
Could you use a shorter placeholder here? like `sa-cockroachdb_` so we avoid 40+ changes?

rafiss replied:
Thanks for the review. If I understand correctly, that's not just placeholder text -- isn't it the text that shows up in the final document? So the link name will be different that the actual name of the dialect package.

I can make this change, but I'm just checking to make sure I understand.

CaselIT replied:
You are right, sorry. I thought that the actual text was provided below, but it seem that the text in the table is indeed what is rendered in the final document.

Since it's more appropriate using the full package name, `sqlalchemy-cockroachdb_` and it requires changing the table anyway, I think we could be a bit generous and have a few more spaces in the right columns

rafiss replied:
That sounds good. Done.

CaselIT replied:
thansk, sorry for the back and forth

rafiss committed:
Use shorter name for sqlalchemy-cockroachdb link

rafiss committed:
Use full name of sqlalchemy-cockroachdb again

This also increases the width of the table so that it's less likely the
whole table will need to be modified again in the future.

zzzeek commented:
merged in 1d12bc138 with squash + cherry-pick and cherry-picked in c0ffe79c0, note the latter pulled in the firebird link also.
